src/index.js

```diff
@@ -104,41 +124,202 @@ class ReactPlayerLoader extends React.Component {
     delete options.attrs;
     delete options.baseUrl;
 
+    // If a base URL is provided, it should only apply to this player load.
+    // This means we need to back up the original base URL and restore it
+    // _after_ we call player loader.
     const originalBaseUrl = playerLoader.getBaseUrl();
 
     if (this.props.baseUrl) {
       playerLoader.setBaseUrl(this.props.baseUrl);
     }
 
     playerLoader(options);
-
     playerLoader.setBaseUrl(originalBaseUrl);
   }
 
   /**
-   * Called just before a component unmounts from the DOM.
-   * Cleans up the player.
+   * Disposes the current player, if there is one.
    */
-  componentWillUnmount() {
-    if (!this.mounted_) {
+  disposePlayer() {
+
+    // Nothing to dispose.
+    if (!this.player) {
       return;
     }
 
-    this.mounted_ = false;
+    // Dispose an in-page player.
+    if (this.player.dispose) {
+      this.player.dispose();
 
-    if (!this.player) {
-      return;
+    // Dispose an iframe player.
+    } else if (this.player.parentNode) {
+      this.player.parentNode.removeChild(this.player);
     }
 
-    disposePlayer(this.player);
+    // Null out the player reference.
     this.player = null;
   }
 
   /**
-   * Renders the player
+   * Update the player based on changes to certain props that do not require
+   * a full player reload.
+   *
+   * @param {Object} changes
+   *        An object. The keys of this object are the props that changed.
+   */
+  updatePlayer(changes) {
+
+    // No player exists or player is disposed.
+    if (!this.player || !this.player.el()) {
+      return;
+    }
+
+    const findPlaylistItemIndex = (items) => {
+      if (Array.isArray(items) && this.props.playlistVideoId) {
+        for (let i = 0; i < items.length; i++) {
+          const item = items[i];
+
+          if (
+            item.id === this.props.playlistVideoId ||
+            `ref:${item.referenceId}` === this.props.playlistVideoId
+          ) {
+            return i;
+          }
+        }
+      }
+      return -1;
+    };
+
+    const catalogCallback = (requestType) => {
```

Comment:
Could this be a helper as well?

Reply:
Moving it to be a private method of the component.